include/benchmark/benchmark.h

```diff
@@ -1418,6 +1424,31 @@ class BENCHMARK_DEPRECATED_MSG("The CSV Reporter will be removed in a future rel
   std::set<std::string> user_counter_names_;
 };
 
+// If a MemoryManager is registered, it can be used to report memory statistics
+// for a run of the benchmark.
+class MemoryManager {
+ public:
+  struct Result {
+    explicit Result(int64_t iters)
+        : num_allocs(0), max_bytes_used(0), iterations(iters) {}
+
+    // The number of allocations made in total between Start and Stop.
+    int64_t num_allocs;
+
+    // The peak memory use between Start and Stop.
+    int64_t max_bytes_used;
+
+    // The number of iterations run for the memory test.
+    const int64_t iterations;
```

Comment:
From what I see in the .cc file, iterations is only used to convey info from one internal function to another.  So maybe this field should not be in the API (header file) and should be just passed as an extra arg internally.  You will end up with two new args in the internal call:
    int64 memory_iterations, const MemoryManager::Result& result

"memory_iterations > 0" can be used to determine whether or not memory usage was recorded, and therefore as a side-benefit, Result will no longer need to be heap-allocated.

Reply:
Done